codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -321,18 +321,19 @@ public InterfaceHttpData getBodyHttpData(String name) {
     public HttpPostMultipartRequestDecoder offer(HttpContent content) {
         checkDestroyed();
 
+        if (content instanceof LastHttpContent) {
+            isLastChunk = true;
+        }
+
+        ByteBuf buf = content.content();
         // Maybe we should better not copy here for performance reasons but this will need
         // more care by the caller to release the content in a correct manner later
         // So maybe something to optimize on a later stage
-        ByteBuf buf = content.content();
         if (undecodedChunk == null) {
-            undecodedChunk = buf.copy();
+            undecodedChunk = isLastChunk ? buf.retainedSlice() : buf.copy();
```

Comment:
@fabienrenaud Can you add a comment that its only fine to not copy here as we will never call `undecodedChunk.writeBytes(...)` later on ?

Reply:
added